server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestCopybookReplacePatterns.java

```diff
@@ -235,4 +245,126 @@ void testCopySuppressReplacing() {
     UseCaseEngine.runTest(
         TEXT10, ImmutableList.of(new CobolText(REPL9_NAME, REPL9)), ImmutableMap.of());
   }
+
+  /**
+   * SYSPRINT OUTPUT: 000008 COPY ATHITS 000009 REPLACING 'XXXX' BY XYZ. 000010C 16105A 01
+   * XYZ-BASE-RCRD-TX PIC X(421). 000000000 421C
+   */
+  public static final String TEXT11 =
+      "       IDENTIFICATION DIVISION.\n"
+          + "       PROGRAM-ID.    TEST12.\n"
+          + "       ENVIRONMENT DIVISION.\n"
+          + "       CONFIGURATION SECTION.\n"
+          + "       INPUT-OUTPUT SECTION.\n"
+          + "       DATA DIVISION.\n"
+          + "       WORKING-STORAGE SECTION.\n"
+          + "       COPY {~ATHITS}\n"
+          + "           REPLACING 'XXXX' BY XYZ.\n"
+          + "       PROCEDURE DIVISION.\n"
+          + "           DISPLAY {$XYZ-BASE-RCRD-TX}.\n"
+          + "           STOP RUN.";
+
+  public static final String ATHITS_NAME = "ATHITS";
+  public static final String ATHITS =
+      "16105A 01  {$*'XXXX'-BASE-RCRD-TX^XYZ-BASE-RCRD-TX}                  PIC X(421).\n";
```

Comment:
Replacing doesnt work if the string literal is not present at the start of the match. For example

```
000006                COPY ATHITS
   000007                    REPLACING  'XXXX' BY XYZ.
   000008C                 01  RCRD-TX-'XXXX'                   PIC X(421).                         

   000006                COPY ATHITS
   000007                    REPLACING  'XXXX' BY XYZ.
   000008C                 01  RCRD-'XXXX'-TX                   PIC X(421).                                  000000000 421C
```

Reply:
```
     000008 COPY ATHITS 000009 REPLACING 'XXXX' BY XYZ. 
     000010C 16105A 01 XYZ-BASE-RCRD-TX PIC X(421).                                                             000000000 421C
```